compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1644,8 +1647,7 @@ private BType createAccumulatedErrorTypeForMatchingRecive(DiagnosticPos pos, BTy
 
     @Override
     public void visit(BLangWorkerSyncSendExpr syncSendExpr) {
-        BSymbol receiver = symResolver.lookupSymbol(env,
-                names.fromIdNode(syncSendExpr.workerIdentifier), SymTag.VARIABLE);
+        BSymbol receiver = symResolver.lookupMainSpaceSymbol(env, names.fromIdNode(syncSendExpr.workerIdentifier));
```

Comment:
We need to add SymTag validation here.

Reply:
Tested the above example with this change, and it throws above error